twine/utils.py

```diff
@@ -34,7 +34,20 @@
 DEFAULT_REPOSITORY = "https://upload.pypi.org/legacy/"
 TEST_REPOSITORY = "https://test.pypi.org/legacy/"
 
-DEFAULT_CONFIG_FILE = "~/.pypirc"
+# Obtain config file location following XDG spec
+XDG_CONFIG_HOME = os.environ.get(
+    "XDG_CONFIG_HOME", os.path.join(os.path.expanduser("~"), ".config")
+)
+XDG_CONFIG_FILE = os.path.join(XDG_CONFIG_HOME, "pypi", "pypirc")
```

Comment:
I should have noticed this in your suggestion on the PR: I don't think Twine should unilaterally decide to put its configuration in a `pypi` directory. First, I don't think it's accurate; despite the name, `.pypirc` can be used to define other package repositories besides PyPI. It could be `twine`, but I think it's also worth keeping in mind that other tools ([like flit](https://flit.readthedocs.io/en/latest/upload.html)) can use `.pypirc`. So maybe it should be be `pypa`, but I'd want buy-in from other PyPA maintainers before claiming that name.

Reply:
@pradyunsg I'm particularly interested in your thoughts from pip's perspective.

Maybe also @takluyver, with flit's perspective.